src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,367 @@
+import OpenLinkIcon from 'src/icons/OpenLinkIcon'
+import { navigate, navigateBack } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import { StackParamList } from 'src/navigator/types'
+import { Nft } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import Logger from 'src/utils/Logger'
+import networkConfig from 'src/web3/networkConfig'
+
+const TAG = 'nfts/NftsInfoCarousel'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.NftsInfoCarousel>
+
+const scaleImageHeight = (originalWidth: number, originalHeight: number, targetWidth: number) => {
+  const aspectRatio = originalWidth / originalHeight
+  return targetWidth / aspectRatio
+}
+
+const NftsLoadErrorScreen = () => {
+  const { t } = useTranslation()
+  return (
+    <SafeAreaView
+      testID="NftsInfoCarousel/NftsLoadErrorScreen"
+      style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}
+    >
+      <Text>{t('nftInfoCarousel.noNftsFound')}</Text>
+    </SafeAreaView>
+  )
+}
+
+const ThumbnailImagePlaceholder = () => {
+  return (
+    <SkeletonPlaceholder backgroundColor={colors.gray2} highlightColor={colors.white}>
+      <View style={styles.skeletonImageThumbnailPlaceHolder} />
+    </SkeletonPlaceholder>
+  )
+}
+
+const NftThumbnail = ({
+  nft,
+  setActiveNft,
+  activeNft,
+}: {
+  nft: Nft
+  setActiveNft: any
+  activeNft: Nft
+}) => {
+  const [loading, setLoading] = useState(true)
+  return (
+    <Touchable borderless={false} onPress={() => setActiveNft(nft)}>
+      <FastImage
+        style={[
+          styles.nftPreviewImageShared,
+          activeNft.tokenUri === nft.tokenUri
+            ? styles.nftPreviewImageSelected
+            : styles.nftPreviewImageUnSelected,
+        ]}
+        source={{
+          uri: nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway,
+        }}
+        onError={() => {
+          Logger.error(TAG, 'Error loading Nft preview image')
+        }}
+        resizeMode={FastImage.resizeMode.cover}
+        onLoadEnd={() => {
+          setLoading(false)
+        }}
+      >
+        {loading && <ThumbnailImagePlaceholder />}
+      </FastImage>
+    </Touchable>
+  )
+}
+
+export default function NftsInfoCarousel({ route }: Props) {
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState((nfts && nfts[0]) ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(360)
+  const { t } = useTranslation()
+
+  const android_ripple = {
+    color: colors.gray2,
+    foreground: true,
+    borderless: true,
+  }
+
+  // Some components that require parent state defined
+  const PlatformSpecificTopBarButtons = () => {
```

Comment:
i also feel like this is a weird place for this component, this seems like it should be in the components folder. what's the diff between the two buttons? (like what happens if we use the `TopBarIconButton` on android, since we already do this pretty much across the app?